### DISCUSSION TITLE
How does the MeatPi gets the 12V battery values for the Sleep Voltage

### DISCUSSION THREAD

**Discussion by stwa2020** [meatpiHQ/wican-fw#136] [Q&A]
Hi All,

Does anyone know how the MeatPi gets the 12V battery values, does it measure the voltage on the ODB2 connector via pin 16 **V+** and pin 5 **GND** or does it keep the CAN BUS running and request the battery values via ELM ?
How much mA are consumed by the MeatPi ?

I ask this question because I have noticed that when my Skoda Enyaq sits for a few hours and the high voltage battery charging starts, the MeatPi stays offline (no wifi) and I don't know if the 12V voltage is too low and it stays off.
I have now changed the sleep voltage from 12.5V to 12V to see if that makes a difference, otherwise I need to measure the mA on the 12V battery when the car is off for a while and see what happens when the high voltage battery starts charging.

Thanks and best regards, 

Stefan

**Comment by meatpiHQ** [comment]
@stwa2020 it's measures the voltage on pin 16, that's the battery voltage. In normal mode it consumes about 25mA. 

I think 12.5V is low enough, does it go to sleep? 12V means that the battery is low.

**Reply by stwa2020** [reply]
I changed my script to request the BMS value via CAN only when the wallbox changes the Wh summary value, now the bus can turn off and the 12V changed from 12.0V to 12.5V and after 12h it changed from 12.5V to 12.4V

I use the following way to get the battery voltage output to use it in IOBroker. I send a http request to `http://meatpi-ip/check_status` this give me an output which includes the battery value` ..cut.. "batt_voltage": "12.3V"..cut..` Now I can parse this value with a script and use it for my automation.

I used blockly and this is the test script that gets the battery information from the HTTP request. (blocky is in german)
Instead of debug link it to a object,

![image](https://github.com/meatpiHQ/wican-fw/assets/66866558/f80968f7-e8ab-4389-8238-984979614d79)
